packages/otelbin/src/components/monaco-editor/Editor.tsx

```diff
@@ -77,7 +78,9 @@ export default function Editor({ locked, setLocked }: { locked: boolean; setLock
 		totalValidationErrors.jsYamlError == null && (totalValidationErrors.ajvErrors?.length ?? 0) === 0;
 
 	const handleEditorChange: OnChange = (value) => {
-		setCurrentConfig(value || "");
+		isK8sConfigMap(value ?? "")
+			? editorRef?.current?.getModel()?.setValue(extractRelayFromK8sConfigMap(value ?? "") ?? "")
```

Comment:
**Question:** `extractRelayFromK8sConfigMap` internally also calls `isK8sConfigMap`. This means that we are currently parsing the content up to three times. This seems a bit extreme. Can we make this a bit more efficient?